File: Nx3CS/bin/test/src/nx3/PNoteheadsRectsCalculator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PNoteheadsRectsCalculator : global::haxe.lang.HxObject {
		public    PNoteheadsRectsCalculator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PNoteheadsRectsCalculator(global::nx3.PNote note){
			unchecked {
				#line 17 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				global::nx3.PNoteheadsRectsCalculator.__hx_ctor_nx3_PNoteheadsRectsCalculator(this, note);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PNoteheadsRectsCalculator(global::nx3.PNoteheadsRectsCalculator __temp_me108, global::nx3.PNote note){
			unchecked {
				#line 19 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				__temp_me108.note = note;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				return new global::nx3.PNoteheadsRectsCalculator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				return new global::nx3.PNoteheadsRectsCalculator(((global::nx3.PNote) (arr[0]) ));
			}
			#line default
		}
		
		
		public  global::nx3.PNote note;
		
		public  global::Array<object> heads;
		
		public  global::Array<object> placements;
		
		public virtual   global::Array<object> getHeadsRects(){
			unchecked {
				#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				{
					#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
					global::nx3.ENoteType _g = this.note.nnote.type;
					#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
					switch (global::Type.enumIndex(_g)){
						case 0:
						{
							#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							global::Array<object> a2 = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (_g.@params[3]) ))) );
							#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							global::Array<global::nx3.ENoteArticulation> a = ((global::Array<global::nx3.ENoteArticulation>) (global::Array<object>.__hx_cast<global::nx3.ENoteArticulation>(((global::Array) (_g.@params[2]) ))) );
							#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							global::nx3.ENotationVariant v = ((global::nx3.ENotationVariant) (_g.@params[1]) );
							#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							global::Array<object> h = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (_g.@params[0]) ))) );
							#line 28 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							return new global::nx3.PHeadsRectsCalculator(((global::nx3.PNote) (this.note) ), ((global::nx3.EDirectionUD) (default(global::nx3.EDirectionUD)) )).getHeadsRects();
						}
						
						
						case 1:
						{
							#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							int l = ((int) (global::haxe.lang.Runtime.toInt(_g.@params[0])) );
							#line 29 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							return new global::nx3.PNoteHeadsRectsPausesCalculator(((global::nx3.PNote) (this.note) )).getHeadsRects();
						}
						
						
						case 4:
						{
							#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							object font = _g.@params[3];
							#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							global::nx3.ELyricContinuation c = ((global::nx3.ELyricContinuation) (_g.@params[2]) );
							#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							global::nx3.EPosition o = ((global::nx3.EPosition) (_g.@params[1]) );
							#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							string text = global::haxe.lang.Runtime.toString(_g.@params[0]);
							#line 30 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							return new global::nx3.PNoteHeadsRectsLyricsCalculator(((global::nx3.PNote) (this.note) ), ((string) (text) ), ((object) (font) )).getHeadsRects();
						}
						
						
						case 3:
						{
							#line 26 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							int l1 = ((int) (global::haxe.lang.Runtime.toInt(_g.@params[0])) );
							#line 31 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							return new global::nx3.PNoteHeadsRectTplCalculator(((global::nx3.PNote) (this.note) )).getHeadsRects();
						}
						
						
						default:
						{
							#line 33 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
							throw global::haxe.lang.HaxeException.wrap(global::haxe.lang.Runtime.concat("Non implemented ENoteType: ", global::Type.enumConstructor(this.note.nnote.type)));
						}
						
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				switch (hash){
					case 1193672430:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						this.placements = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (@value) ))) );
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						return @value;
					}
					
					
					case 616092499:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						this.heads = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (@value) ))) );
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						return @value;
					}
					
					
					case 1225398258:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						this.note = ((global::nx3.PNote) (@value) );
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				switch (hash){
					case 325402290:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getHeadsRects") ), ((int) (325402290) ))) );
					}
					
					
					case 1193672430:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						return this.placements;
					}
					
					
					case 616092499:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						return this.heads;
					}
					
					
					case 1225398258:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						return this.note;
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				switch (hash){
					case 325402290:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						return this.getHeadsRects();
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				baseArr.push("placements");
				#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				baseArr.push("heads");
				#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				baseArr.push("note");
				#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
				{
					#line 11 "F:\\nx3\\src\\nx3\\PNoteheadsRectsCalculator.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Nx3Java/java/src/nx3/PNoteheadsRectsCalculator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PNoteheadsRectsCalculator extends haxe.lang.HxObject
{
	public    PNoteheadsRectsCalculator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PNoteheadsRectsCalculator(nx3.PNote note)
	{
		nx3.PNoteheadsRectsCalculator.__hx_ctor_nx3_PNoteheadsRectsCalculator(this, note);
	}
	
	
	public static   void __hx_ctor_nx3_PNoteheadsRectsCalculator(nx3.PNoteheadsRectsCalculator __temp_me125, nx3.PNote note)
	{
		__temp_me125.note = note;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PNoteheadsRectsCalculator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PNoteheadsRectsCalculator(((nx3.PNote) (arr.__get(0)) ));
	}
	
	
	public  nx3.PNote note;
	
	public  haxe.root.Array<nx3.PHead> heads;
	
	public  haxe.root.Array<java.lang.Object> placements;
	
	public   haxe.root.Array<nx3.geom.Rectangle> getHeadsRects()
	{
		{
			nx3.ENoteType _g = this.note.nnote.type;
			switch (haxe.root.Type.enumIndex(_g))
			{
				case 0:
				{
					haxe.root.Array<nx3.ENoteAttributes> a2 = ((haxe.root.Array<nx3.ENoteAttributes>) (_g.params.__get(3)) );
					haxe.root.Array<nx3.ENoteArticulation> a = ((haxe.root.Array<nx3.ENoteArticulation>) (_g.params.__get(2)) );
					nx3.ENotationVariant v = ((nx3.ENotationVariant) (_g.params.__get(1)) );
					haxe.root.Array<nx3.NHead> h = ((haxe.root.Array<nx3.NHead>) (_g.params.__get(0)) );
					return new nx3.PHeadsRectsCalculator(((nx3.PNote) (this.note) ), ((nx3.EDirectionUD) (null) )).getHeadsRects();
				}
				
				
				case 1:
				{
					int l = ((int) (haxe.lang.Runtime.toInt(_g.params.__get(0))) );
					return new nx3.PNoteHeadsRectsPausesCalculator(((nx3.PNote) (this.note) )).getHeadsRects();
				}
				
				
				case 4:
				{
					java.lang.Object font = _g.params.__get(3);
					nx3.ELyricContinuation c = ((nx3.ELyricContinuation) (_g.params.__get(2)) );
					nx3.EPosition o = ((nx3.EPosition) (_g.params.__get(1)) );
					java.lang.String text = haxe.lang.Runtime.toString(_g.params.__get(0));
					return new nx3.PNoteHeadsRectsLyricsCalculator(((nx3.PNote) (this.note) ), haxe.lang.Runtime.toString(text), ((java.lang.Object) (font) )).getHeadsRects();
				}
				
				
				case 3:
				{
					int l1 = ((int) (haxe.lang.Runtime.toInt(_g.params.__get(0))) );
					return new nx3.PNoteHeadsRectTplCalculator(((nx3.PNote) (this.note) )).getHeadsRects();
				}
				
				
				default:
				{
					throw haxe.lang.HaxeException.wrap(( "Non implemented ENoteType: " + haxe.root.Type.enumConstructor(this.note.nnote.type) ));
				}
				
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef732 = true;
			switch (field.hashCode())
			{
				case -253474258:
				{
					if (field.equals("placements")) 
					{
						__temp_executeDef732 = false;
						this.placements = ((haxe.root.Array<java.lang.Object>) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 3387378:
				{
					if (field.equals("note")) 
					{
						__temp_executeDef732 = false;
						this.note = ((nx3.PNote) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 99151507:
				{
					if (field.equals("heads")) 
					{
						__temp_executeDef732 = false;
						this.heads = ((haxe.root.Array<nx3.PHead>) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef732) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef733 = true;
			switch (field.hashCode())
			{
				case 2051530802:
				{
					if (field.equals("getHeadsRects")) 
					{
						__temp_executeDef733 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getHeadsRects"))) );
					}
					
					break;
				}
				
				
				case 3387378:
				{
					if (field.equals("note")) 
					{
						__temp_executeDef733 = false;
						return this.note;
					}
					
					break;
				}
				
				
				case -253474258:
				{
					if (field.equals("placements")) 
					{
						__temp_executeDef733 = false;
						return this.placements;
					}
					
					break;
				}
				
				
				case 99151507:
				{
					if (field.equals("heads")) 
					{
						__temp_executeDef733 = false;
						return this.heads;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef733) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef734 = true;
			switch (field.hashCode())
			{
				case 2051530802:
				{
					if (field.equals("getHeadsRects")) 
					{
						__temp_executeDef734 = false;
						return this.getHeadsRects();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef734) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("placements");
		baseArr.push("heads");
		baseArr.push("note");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


